Editor/PostBuild.cs

```diff
@@ -1,130 +1,229 @@
+
+		var fileExtensions = new List<string>()
+		{
+			".dll",
+			".pdb"
+		};
+		var symbolFiles = Directory.GetFiles(pluginsDir, "*", SearchOption.AllDirectories)
+			.Select(file => new FileInfo(file))
+			.Where(fileInfo => fileExtensions.Any(ext => ext.Equals(fileInfo.Extension)))
+			.ToList();
+
+		foreach (var symbolFile in symbolFiles)
+		{
+			Debug.Log($"BugSplat found symbol file: {symbolFile.FullName}");
+		}
+
+		Debug.Log("About to upload symbol files to BugSplat...");
+
+		try
+		{
+			using (var symbolUploader = SymbolUploader.CreateOAuth2SymbolUploader(clientId, clientSecret))
+			{
+				var responseMessages = await symbolUploader.UploadSymbolFiles(
+					database,
+					application,
+					version,
+					symbolFiles
+				);
+
+				if (responseMessages[0].IsSuccessStatusCode)
+					Debug.Log("BugSplat symbol upload completed successfully!");
+				else
+					Debug.LogError("BugSplat symbol upload failed. " + responseMessages[0]);
+			}
+		}
+		catch (Exception ex)
+		{
+			Debug.LogError(ex);
+		}
+	}
+
+	private static BugSplatOptions GetBugSplatOptions()
+	{
+		var guids = AssetDatabase.FindAssets("t:BugSplatOptions");
+
+		if (guids.Length == 0)
+		{
+			return null;
+		}
+
+		var path = AssetDatabase.GUIDToAssetPath(guids[0]);
+		return AssetDatabase.LoadAssetAtPath<BugSplatOptions>(path);
+	}
+	
+#if UNITY_IOS
+	private static void PostProcessIos(string pathToBuiltProject, BugSplatOptions options)
+	{
+		var projectPath = PBXProject.GetPBXProjectPath(pathToBuiltProject);
+
+		var project = new PBXProject();
+		project.ReadFromString(File.ReadAllText(projectPath));
+
+#if UNITY_2019_3_OR_NEWER
+		var targetGuid = project.GetUnityFrameworkTargetGuid();
+#else
+		var targetName = PBXProject.GetUnityTargetName();
+		var targetGuid = project.TargetGuidByName(targetName);
+#endif
+
+		project.AddBuildProperty(targetGuid, "OTHER_LDFLAGS", "-ObjC");
+		project.AddBuildProperty(targetGuid, "ENABLE_BITCODE", "NO");
+
+		project.SetBuildProperty(targetGuid, "DEBUG_INFORMATION_FORMAT", "dwarf-with-dsym");
+
+		var mainTargetGuid = project.GetUnityMainTargetGuid();
+		project.AddBuildProperty(mainTargetGuid, "ENABLE_BITCODE", "NO");
+		project.SetBuildProperty(mainTargetGuid, "DEBUG_INFORMATION_FORMAT", "dwarf-with-dsym");
+
+		ModifyPlist(pathToBuiltProject, options);
+		AddBundle(pathToBuiltProject, project, targetGuid);
+		HandleUploadSymbols(mainTargetGuid, project, options.UploadDebugSymbolsForIos);
+
+		File.WriteAllText(projectPath, project.WriteToString());
+	}
+
+	private static void ModifyPlist(string projectPath, BugSplatOptions options)
+	{
+		var plistInfoFile = new PlistDocument();
+
+		var infoPlistPath = Path.Combine(projectPath, "Info.plist");
+		plistInfoFile.ReadFromString(File.ReadAllText(infoPlistPath));
+
+		const string bugSplatServerURLKey = "BugsplatServerURL";
+		plistInfoFile.root.AsDict().SetString(bugSplatServerURLKey, $"https://{options.Database}.bugsplat.com/");
+
```

Comment:
I could use a bit of help understanding our options for version here. I think perhaps this is our hook into getting the versions matching for our symbol uploads. 

* In Unreal I believe we chopped of the build revision (the (3) part in 1.0 (3)).
    * Can we make a similar change in Unity so that there's only one source of truth for the application's version? 
    * If the user builds the native solution again, will this step be re-run and update the version in Info.plist?
* Or should we instruct the user to manage their version number manually?

Any details you can add are greatly appreciated!

Reply:
Yes, the best solution would be to use the application version, as it is already managed by the user in the project settings.

It would be redundant to make the user type the app version in two different places.